eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -4,24 +4,53 @@ jobs:
     matrix:
       Linux_x64:
         vm.image: 'ubuntu-18.04'
-        vcpkg.deps: 'curl[ssl]'
+        vcpkg.deps: ''
         VCPKG_DEFAULT_TRIPLET: 'x64-linux'
+        buid.args: ''
       Win_x86:
         vm.image: 'windows-2019'
-        vcpkg.deps: 'curl[winssl]'
+        vcpkg.deps: ''
         VCPKG_DEFAULT_TRIPLET: 'x86-windows-static'
         CMAKE_GENERATOR: 'Visual Studio 16 2019'
         CMAKE_GENERATOR_PLATFORM: Win32
+        buid.args: ''
       Win_x64:
         vm.image: 'windows-2019'
-        vcpkg.deps: 'curl[winssl]'
+        vcpkg.deps: ''
         VCPKG_DEFAULT_TRIPLET: 'x64-windows-static'
         CMAKE_GENERATOR: 'Visual Studio 16 2019'
         CMAKE_GENERATOR_PLATFORM: x64
+        buid.args: ''
       MacOS_x64:
+        vm.image: 'macOS-10.14'
+        vcpkg.deps: ''
+        VCPKG_DEFAULT_TRIPLET: 'x64-osx'
+        buid.args: ''
+      
+      Linux_x64_with_samples:
+        vm.image: 'ubuntu-18.04'
+        vcpkg.deps: 'curl[ssl]'
+        VCPKG_DEFAULT_TRIPLET: 'x64-linux'
+        buid.args: ' -DBUILD_CURL_TRANSPORT=ON'
```

Comment:
Stylistically, I think the approach of setting build configs as environment variables will make it easier to read what we're doing in the matrix. 

For example I'm doing this in another PR https://github.com/Azure/azure-sdk-for-c/pull/343/files#diff-a36606b54a95a0d62cfd379adb42c11eR32 

This could be written instead as: 
```
      Linux_x64_with_samples:
        vm.image: 'ubuntu-18.04'
        vcpkg.deps: 'curl[ssl]'
        VCPKG_DEFAULT_TRIPLET: 'x64-linux'
        BUILD_CURL_TRANSPORT: ON
```

By contrast, we could provide the variables  to the build using `$(build.args)` in the cmake generate step but over time the `build.args` variable will get horizontally long and could make it more difficult to reason about what configurations are going into the matrix. 

For example:
```
      Linux_x64_with_samples:
        vm.image: 'ubuntu-18.04'
        vcpkg.deps: 'curl[ssl]'
        VCPKG_DEFAULT_TRIPLET: 'x64-linux'
        buid.args: ' -DBUILD_CURL_TRANSPORT=ON -DSOME_OTHER_BUILD_OPTION=ON -DSOME_SOMEWHAT_MORE_OBSCURE_BUILD_OPTION=ON -DDONT_FORGET_TO_SCROLL_RIGHT=ON -DETC=ON'
```

Reply:
This whole comment is wrong. I was confusing environment variables with another one of the build systems I maintain :)

Please ignore.